app/services/submission/plant_trial_finalizer.rb

```diff
@@ -164,22 +164,26 @@ def create_scoring
   end
 
   def create_plant_trial
-    attrs = submission.content.step01.to_h.merge(common_data)
-    design_factor_names = submission.content.step04.design_factor_names || []
+    attr_names = [:plant_trial_name, :project_descriptor, :plant_population_id, :plant_trial_description, :trial_year,
+                  :institute_id, :country_id, :trial_location_site_name, :place_name, :latitude, :longitude, :altitude,
+                  :terrain, :soil_type, :statistical_factors, :data_status]
 
-    if plant_population = PlantPopulation.find_by(id: submission.content.step01.plant_population_id)
+    attrs = submission.content.to_h.slice(*attr_names).merge(common_data)
```

Comment:
In `create_plant_population` you merge submission content to common_data. Here, you proceed in a reverse fashion. Any particular reason?

Reply:
Not really, it's because of the way code was structured before these changes.